backend-lucene/src/main/java/org/hibernate/search/v6poc/backend/lucene/document/impl/AbstractLuceneDocumentBuilder.java

```diff
@@ -36,9 +36,7 @@ protected AbstractLuceneDocumentBuilder(LuceneIndexSchemaObjectNode schemaNode)
 	}
 
 	@Override
-	public void addNestedObjectDocumentBuilder(LuceneIndexSchemaObjectNode expectedParentNode, LuceneNestedObjectDocumentBuilder nestedObjectDocumentBuilder) {
-		checkTreeConsistency( expectedParentNode );
-
+	public void addNestedObjectDocumentBuilder(LuceneNestedObjectDocumentBuilder nestedObjectDocumentBuilder) {
```

Comment:
Wondering about this one as we have the exact same pattern in Elasticsearch. Not sure what to do about it. I suppose you don't have any cyclic dependency or you would have fixed it too but I find it a bit annoying to have 2 different patterns.

Reply:
Ok, I'll do the same in Elasticsearch.
It's just that I didn't need to do anything in the ES integration because, there, the formatter is not responsible for adding fields, just for converting data, and the field accessor/document builder themselves take care of adding fields.

Reply:
Thanks, I think consistency is for the best in this case, even if not strictly needed.